### PR TITLE
Add 2.7 blog post

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

https://deploy-preview-12966--prettier.netlify.app/blog/2022/06/11/2.7.0.html

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: changelog_unreleased/blog-post-intro.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+author: "Sosuke Suzuki (@sosukesuzuki)"
+authorURL: "https://github.com/sosukesuzuki"
+title: "Prettier 2.7: new --cache CLI option and TypeScript 4.7 syntax!"
+---
+
+This release includes a new `--cache` CLI option. Enabling this option will use some attributes as cache keys and format files only if they have changed. This could dramatically improve CLI performance.
```
Comment (sosukesuzuki) [inline changelog_unreleased/blog-post-intro.md:7]: #12800 has not been merged yet.

File: website/blog/2022-06-04-2.7.0.md [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+  dogStuff: any;
+}
+type Getter<out T> = () => T;
+type Setter<in T> = (value: T) => void;
+```
+
+##### [`extends` constraints for `infer`](https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#extends-constraints-on-infer-type-variables)
+
+```ts
+type FirstString<T> = T extends [infer extends string, ...unknown[]]
```
Comment (Tiedye) [inline website/blog/2022-06-04-2.7.0.md:48]:
```suggestion
type FirstString<T> = T extends [infer S extends string, ...unknown[]]
```

Comment (sosukesuzuki) [reply website/blog/2022-06-04-2.7.0.md]: Thank you! [ee06ea9](https://github.com/prettier/prettier/pull/12966/commits/ee06ea9fc1464ded8a8e4bcbda022923743a08cf)

File: changelog_unreleased/blog-post-intro.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+We've also added support formatting for TypeScript 4.7 syntax!
```
Comment (thebanjomatic) [inline changelog_unreleased/blog-post-intro.md:9]:
```suggestion
We've also added support for formatting  TypeScript 4.7 syntax!
```

File: website/blog/2022-06-10-2.7.0.md [hunk 4]
```diff
@@ -0,0 +1,466 @@
+---
+author: "Sosuke Suzuki (@sosukesuzuki)"
+authorURL: "https://github.com/sosukesuzuki"
+title: "Prettier 2.7: new --cache CLI option and TypeScript 4.7 syntax!"
+---
+
+This release includes a new `--cache` CLI option. Enabling this option will use some attributes as cache keys and format files only if they have changed. This could dramatically improve CLI performance.
+
+We've also added support formatting for TypeScript 4.7 syntax!
```
Comment (thebanjomatic) [inline website/blog/2022-06-10-2.7.0.md:9]:
```suggestion
We've also added support for formatting TypeScript 4.7 syntax!
```

File: website/blog/2022-06-11-2.7.0.md [hunk 5]
```diff
@@ -0,0 +1,513 @@
...
+  value: T;
+}
+function makeBox<T>(value: T) {
+  return { value };
+}
+const makeHammerBox = makeBox<Hammer>;
+const makeWrenchBox = makeBox<Wrench>;
+```
+
+##### [Optional Variance](https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#optional-variance-annotations-for-type-parameters)
```
Comment (Josh-Cena) [inline website/blog/2022-06-11-2.7.0.md:35]:
```suggestion
##### [Optional Variance Annotations](https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#optional-variance-annotations-for-type-parameters)
```

File: website/blog/2022-06-11-2.7.0.md [hunk 6]
```diff
@@ -0,0 +1,513 @@
...
+test.describe
+  .serial("does something really long and complicated so I have to write a very long name for the test", () => {});
+
+test.describe.serial
+  .only("does something really long and complicated so I have to write a very long name for the test", () => {});
+```
+
+#### Fix comment formats in ([#12860](https://github.com/prettier/prettier/pull/12860) by [@HosokawaR](https://github.com/HosokawaR))
+
+This change fixes comments format in `exports` to algin with the comment format in `import`
```
Comment (Josh-Cena) [inline website/blog/2022-06-11-2.7.0.md:206]:
```suggestion
This change fixes the comment format in `exports` to align with the comment format in `import`.
```

File: website/blog/2022-06-11-2.7.0.md [hunk 7]
```diff
@@ -0,0 +1,513 @@
...
+enum A {
+  [i++],
+}
+```
+
+### HTML
+
+#### Support Speculation Rules API formatting in HTML ([#12882](https://github.com/prettier/prettier/pull/12882) by [@sosukesuzuki](https://github.com/sosukesuzuki))
+
+Please read https://web.dev/speculative-prerendering/ for about Speculation Rules API.
```
Comment (Josh-Cena) [inline website/blog/2022-06-11-2.7.0.md:294]:
```suggestion
Please read https://web.dev/speculative-prerendering/ for more information about the Speculation Rules API.
```

File: website/blog/2022-06-11-2.7.0.md [hunk 8]
```diff
@@ -0,0 +1,513 @@
...
+<!-- prettier-ignore -->
+```ts
+type FirstString<T> = T extends [infer S extends string, ...unknown[]]
+  ? S
+  : never;
+```
+
+### CLI
+
+#### Add `--cache` and `--cache-strategy` CLI option ([#12800](https://github.com/prettier/prettier/pull/12800) by [@sosukesuzuki](https://github.com/sosukesuzuki))
```
Comment (Josh-Cena) [inline website/blog/2022-06-11-2.7.0.md:60]:
```suggestion
#### Add `--cache` and `--cache-strategy` CLI options ([#12800](https://github.com/prettier/prettier/pull/12800) by [@sosukesuzuki](https://github.com/sosukesuzuki))
```

File: website/blog/2022-06-11-2.7.0.md [hunk 9]
```diff
@@ -0,0 +1,513 @@
...
+
+test.describe.serial
+  .only("does something really long and complicated so I have to write a very long name for the test", () => {});
+```
+
+#### Fix comment formats in ([#12860](https://github.com/prettier/prettier/pull/12860) by [@HosokawaR](https://github.com/HosokawaR))
+
+This change fixes the comment format in `exports` to align with the comment format in `import`.
+
+Although this change does not affect the comments format in `import`, follows change log contains examples of comments in `import` for reference.
```
Comment (Josh-Cena) [inline website/blog/2022-06-11-2.7.0.md:208]:
```suggestion
Although this change does not affect the comments format in `import`, the following changeset contains examples of comments in `import` for reference.
```

(Skeptical about either "change log" or "changeset" being the right word, but nothing better on my mind)

File: website/blog/2022-06-11-2.7.0.md [hunk 10]
```diff
@@ -0,0 +1,513 @@
...
+
+$ prettier src --check
+Checking formatting...
+[warn] src/fileA.js
+[warn] Code style issues found in the above file. Forgot to run Prettier?
+```
+
+#### Infer parser for `.importmap` files ([#12603](https://github.com/prettier/prettier/pull/12603) by [@fisker](https://github.com/fisker))
+
+Format [`.importmap`](https://github.com/WICG/import-maps#import-map-processing) files as a JSON files.
```
Comment (fisker) [inline website/blog/2022-06-11-2.7.0.md:506]:
```diff
Format [`.importmap`](https://github.com/WICG/import-maps#import-map-processing) files as JSON files.
```

**Review by fisker — APPROVED** [review]

**Pull request merged**